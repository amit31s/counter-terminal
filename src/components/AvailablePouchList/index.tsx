import { PouchAcceptanceDetails } from "@ct/api/generator";
import { useGetPouchForAcceptance } from "@ct/features/PouchAcceptanceFeature/useGetPouchForAcceptance";
import { priceToRender } from "@ct/utils";
import { useMemo, useState } from "react";
import { FlatList, Text, TouchableHighlight, View } from "react-native";
import { stringConstants } from "../../constants";
import colorConstants from "../../constants/ColorConstants";
import styles from "./styles";

interface AvailablePouchListProps {
  availablePouches: PouchAcceptanceDetails[];
  parentCallback: (pouch: PouchAcceptanceDetails) => void;
  pouchValueAddition?: string;
}

const AvailablePouchList = ({ availablePouches, parentCallback }: AvailablePouchListProps) => {
  const { validatedData } = useGetPouchForAcceptance();

  availablePouches = useMemo(() => {
    const validatedBarcodes = validatedData.map((validatedPouch) => validatedPouch.pouchID);
    if (!availablePouches || !Array.isArray(availablePouches)) {
      return [];
    }
    return availablePouches.filter(
      (availablePouch) => !validatedBarcodes.includes(availablePouch.pouchID),
    );
  }, [availablePouches, validatedData]);

  const [selectedId, setSelectedId] = useState<string | null>(null);
  const touchProps = {
    activeOpacity: 1,
    underlayColor: colorConstants.disabledTextColour,
  };

  const handleClick = (item: PouchAcceptanceDetails) => {
    if (!item.pouchID) {
      return;
    }
    setSelectedId(item.pouchID);
    parentCallback(item);
  };

  const renderItem = ({ item, index }: { item: PouchAcceptanceDetails; index: number }) => {
    return (
      <TouchableHighlight
        {...touchProps}
        testID={`post-row-${index}`}
        onPress={() => handleClick(item)}
        style={[styles.boxView, styles.tableRow]}
      >
        <View style={styles.flexRow}>
          <View style={styles.width50}>
            <Text style={styles.primaryText}>{item.pouchID}</Text>
          </View>
          <View style={[styles.width50, styles.pouchType]}>
            <Text style={[styles.primaryText, styles.textAlignCenter, styles.pouchTypeValue]}>
              {item.pouchType}
            </Text>
          </View>
          <View style={styles.width50}>
            <Text style={[styles.primaryText, styles.textAlignRight]}>
              {priceToRender(Math.abs(item?.totalValue ?? 0))}
            </Text>
          </View>
        </View>
      </TouchableHighlight>
    );
  };

  const renderHeader = () => {
    return (
      <View style={[styles.tableRow, styles.boxView, styles.flexRow]}>
        <View style={styles.width50}>
          <Text style={styles.barcodeTxtText}>{stringConstants.Pouch.BarCode}</Text>
        </View>
        <View style={styles.width50}>
          <Text style={[styles.tableRowHeaderText, styles.textAlignCenter]}>Pouch Type</Text>
        </View>
        <View style={styles.width50}>
          <Text style={[styles.tableRowHeaderText, styles.textAlignRight]}>
            {stringConstants.Pouch.PouchValue}
          </Text>
        </View>
      </View>
    );
  };

  return (
    <View>
      {renderHeader()}
      <FlatList
        testID="Body"
        data={availablePouches}
        renderItem={renderItem}
        keyExtractor={(item) => item.pouchID || item.pouchID || ""}
        extraData={selectedId}
      />
    </View>
  );
};

export default AvailablePouchList;
