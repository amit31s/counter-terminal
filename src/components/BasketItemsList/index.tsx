export { default as BasketItemsList } from "./BasketItemsList";
