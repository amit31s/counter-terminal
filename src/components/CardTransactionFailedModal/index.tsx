export { default as CardTransactionFailedModal } from "./CardTransactionFailedModal";
