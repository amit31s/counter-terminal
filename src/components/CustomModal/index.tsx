export { default as CustomModal } from "./CustomModal";
export * from "./CustomModal";
