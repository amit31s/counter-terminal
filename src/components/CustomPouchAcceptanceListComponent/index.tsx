import { PouchAcceptanceDetails } from "@ct/api/generator";
import { StyledDeleteIcon } from "@ct/assets/icons";
import { priceToRender } from "@ct/utils";
import { FlatList, Pressable, Text, View } from "react-native";
import stringConstants from "../../constants/StringsConstants";
import styles from "./style";

interface CustomPouchAcceptanceListComponentProps {
  parentdata: PouchAcceptanceDetails[];
  showDelete?: boolean;
  onDeleteItem?: (barcode: string) => void;
}

interface ItemViewProps {
  index: number;
  item: PouchAcceptanceDetails;
}

const ListHeader = ({ showDelete }: { showDelete: boolean }) => {
  return (
    <View style={[styles.textContainer, styles.headerFooterStyle, styles.border]}>
      <Text style={[styles.primaryText, styles.textStyle, styles.textAlignLeft]}>
        {stringConstants.Pouch.BarCode}
      </Text>
      {showDelete ? (
        <Text style={[styles.primaryText, styles.textStyle, styles.textAlignLeft]}>
          {stringConstants.Pouch.PouchType}
        </Text>
      ) : (
        <Text style={[styles.primaryText, styles.textStyle]}>
          {stringConstants.Pouch.PouchType}
        </Text>
      )}
      <Text style={[styles.primaryText, styles.textStyle, styles.pouchValue]}>
        {stringConstants.Pouch.PouchValue}
      </Text>
      {showDelete ? (
        <Text style={[styles.primaryText, styles.textStyle]}>{stringConstants.Pouch.Branch}</Text>
      ) : null}
      {showDelete ? <Text style={[styles.primaryText, styles.actionStyle]}>{""}</Text> : null}
    </View>
  );
};

const EmptyListMessage = () => <Text style={styles.emptyListStyle}>No Data Found</Text>;

const CustomPouchAcceptanceListComponent = ({
  parentdata,
  onDeleteItem,
  showDelete = true,
}: CustomPouchAcceptanceListComponentProps) => {
  const itemView = ({ item, index }: ItemViewProps) => {
    return (
      <View style={[styles.textContainer, styles.border]} testID={`pouch-row-${index}`}>
        <Text style={[styles.primaryText, styles.itemStyle, styles.textAlignLeft]}>
          {item.pouchID}
        </Text>
        <Text style={[styles.primaryText, styles.itemStyle, styles.pouchTypeValue]}>
          {item.pouchType}
        </Text>
        <Text style={[styles.primaryText, styles.itemStyle, styles.pouchValue]}>
          {priceToRender(Math.abs(item?.totalValue ?? 0))}
        </Text>

        {showDelete ? (
          <>
            <Text style={[styles.primaryText, styles.itemStyle]}>{item.assignedBranchName}</Text>
            <Pressable
              onPress={() => {
                if (onDeleteItem && item.pouchID) {
                  onDeleteItem(item.pouchID);
                }
              }}
              style={styles.actionStyle}
            >
              <StyledDeleteIcon />
            </Pressable>
          </>
        ) : null}
      </View>
    );
  };

  return (
    <FlatList
      testID="testFlatList"
      contentContainerStyle={styles.contentContainerStyle}
      data={parentdata}
      keyExtractor={(item: PouchAcceptanceDetails, index: number) =>
        item.pouchID || index.toString()
      }
      ListHeaderComponent={<ListHeader showDelete={showDelete} />}
      renderItem={itemView}
      ListEmptyComponent={EmptyListMessage}
      stickyHeaderIndices={[0]}
    />
  );
};

export default CustomPouchAcceptanceListComponent;
