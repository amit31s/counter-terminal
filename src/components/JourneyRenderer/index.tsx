import { getJourneyStatus, useAppSelector } from "@ct/common";
import { JOURNEY_LOGS_FN } from "@ct/common/constants/JourneyLogs";
import useKeyboardAvoider from "@ct/common/hooks/useKeyboardAvoider";
import { JourneyInterruptContext } from "@ct/screens/HomeScreen/JourneyInterruptContext";
import { LOGGER_TYPE } from "@pol/frontend-logger-common";
import { logManager } from "@pol/frontend-logger-web";
import { JourneyEngine, ProviderProps } from "postoffice-spm-journey-engine";
import { useCallback, useContext, useMemo, useState } from "react";
import { Dimensions, ScrollView, Text, View } from "react-native";
import { v4 } from "uuid";
import { stringConstants } from "../../constants";
import { enablerConfig } from "./enablerConfig";
import styles from "./style";
import { ReturnTypes, useAppData } from "./useAppData";
import { useDeviceCallback } from "./useDeviceCallback";
import { useLogger } from "./useLogger";
// Keeping as 'any' for Journey onComplete payload data because JourneyData type from Journey engine do not enforce many keys used in basket items basically its any key value based also changes required in CT for using this interface is big and would require a seperate ticket.
export type JourneyData = any;
const journeyEngineLogger = logManager(LOGGER_TYPE.journeyEngineLogger);

interface IJourneyRenderer {
  onJourneyAbort?: () => void;
  journeyOnComplete: (data: JourneyData) => Promise<void>;
  barcodeNotRecognizedCb: () => void;
}

export const JourneyRenderer = ({
  journeyOnComplete,
  onJourneyAbort,
  barcodeNotRecognizedCb,
}: IJourneyRenderer) => {
  const [journeyReturns, setJourneyReturns] = useState<ReturnTypes>();
  const appData = useAppData({ journeyReturns });
  const complianceStatus = true;
  const { keyboardPadding } = useKeyboardAvoider();
  const { session } = useAppSelector((rootState) => rootState.auth);
  const windowHeight = Dimensions.get("window").height / 1.7;
  const { interruptionInputData } = useContext(JourneyInterruptContext);
  const deviceCallback = useDeviceCallback(appData, journeyOnComplete);
  const journeyLogger = useLogger();
  const { open } = useAppSelector(getJourneyStatus);

  const contentContainerStyle = useMemo(
    () => ({ paddingBottom: keyboardPadding }),
    [keyboardPadding],
  );

  const getAuthHeaders = useCallback(
    async () => ({
      Authorization: session?.idToken ?? "",
      "X-Correlation-ID": v4(),
    }),
    [session?.idToken],
  );

  const onComplete: ProviderProps["onComplete"] = useCallback(
    (data) => {
      if (data && data.returns) {
        setJourneyReturns(data.returns);
      }
      journeyOnComplete(data);
    },
    [journeyOnComplete],
  );

  const onAbort: ProviderProps["onAbort"] = useCallback(
    (data) => {
      if (data && data.returns) {
        setJourneyReturns(data.returns);
      }
      onJourneyAbort?.();
    },
    [onJourneyAbort],
  );

  const onError: NonNullable<ProviderProps["onError"]> = useCallback(
    (error) => {
      journeyEngineLogger.error({
        methodName: JOURNEY_LOGS_FN.journeyEngineProvider,
        error: error as unknown as Error,
        data: interruptionInputData,
      });
      if (interruptionInputData?.value) {
        barcodeNotRecognizedCb();
      }
    },
    [barcodeNotRecognizedCb, interruptionInputData],
  );

  if (!complianceStatus) {
    return (
      <View style={styles(open).complianceCont}>
        <Text style={styles(open).compWarning}>
          {stringConstants.messages.outstandingCompliance}
        </Text>
      </View>
    );
  }

  return (
    <ScrollView style={styles(open).journeyContainer} contentContainerStyle={contentContainerStyle}>
      <JourneyEngine.Provider
        enablerConfig={enablerConfig}
        authHeaders={getAuthHeaders}
        appData={appData}
        logger={journeyLogger}
        interruptionInputData={interruptionInputData}
        onDeviceTrigger={deviceCallback}
        onComplete={onComplete}
        onAbort={onAbort}
        onError={onError}
      >
        <JourneyEngine.Canvas canvasMinHeight={windowHeight} />
      </JourneyEngine.Provider>
    </ScrollView>
  );
};

export type IJourneyData = JourneyData;
