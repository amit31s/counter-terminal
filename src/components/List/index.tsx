import { CashLocation } from "@ct/common/state/cashTransfer/cashTransfer.slice";
import { TEXT } from "@ct/constants";
import { useState } from "react";
import { FlatList, Text, TouchableHighlight, View } from "react-native";
import colorConstants from "../../constants/ColorConstants";
import styles from "./styles";

export interface ListProps {
  list: CashLocation[];
  selectedLocationCallback: (item: CashLocation) => void;
}

export default function List({ list, selectedLocationCallback }: ListProps) {
  const listItem: CashLocation[] = list;
  const [selectedId, setSelectedId] = useState<string | null>(null);
  const touchProps = {
    activeOpacity: 1,
    underlayColor: colorConstants.disabledTextColour,
  };

  const handleClick = (item: CashLocation) => {
    if (!item?.accountingLocationID) {
      return;
    }
    setSelectedId(item?.accountingLocationID);
    selectedLocationCallback(item);
  };

  const renderItem = ({ item, index }: { item: CashLocation; index: number }) => {
    const selectedStyle = selectedId === item.accountingLocationID ? styles.buttonPressed : null;
    return (
      <TouchableHighlight
        testID={`post-row-${index}`}
        onPress={() => handleClick(item)}
        style={[styles.boxView, styles.tableRow, selectedStyle]}
        {...touchProps}
      >
        <View style={styles.flexRow}>
          <View style={styles.tableRowWidth1}>
            <Text style={styles.primaryText}>{item.accountingLocationName}</Text>
          </View>
        </View>
      </TouchableHighlight>
    );
  };

  const renderHeader = () => {
    return (
      <View style={[styles.tableRow, styles.boxView, styles.flexRow]}>
        <View style={styles.tableRowWidth1}>
          <Text style={styles.tableRowHeaderText}>{TEXT.CTTXT00065}</Text>
        </View>
      </View>
    );
  };

  return (
    <View>
      {renderHeader()}
      <FlatList
        style={styles.listHeight}
        testID="Body"
        data={listItem}
        renderItem={renderItem}
        keyExtractor={(item) => item.accountingLocationID ?? "undefined"}
        extraData={selectedId}
      />
    </View>
  );
}
