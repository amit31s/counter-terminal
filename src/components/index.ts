export * from "./AvailablePouchList";
export * from "./BasketItemsList";
export * from "./CardTransactionFailedModal";
export * from "./CustomModal";
export * from "./HorizontalRule";
export * from "./LoadingModal";
export * from "./PinPadModal";
export * from "./StyledButton";
export * from "./StyledInput";
export * from "./SubmitPouchModal";
