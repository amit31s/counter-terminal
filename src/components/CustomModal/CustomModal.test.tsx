import { render } from "@ct/common";
import { CustomModal } from "@ct/components";
import { imageConstants, stringConstants } from "@ct/constants";

describe("Custom Modal", () => {
  it("renders anything", async () => {
    const view = render(<CustomModal isOpen={true} content="test" />);
    expect(await view.findByText("test")).toBeTruthy();
  });

  it("renders a title when asked", async () => {
    const { getByTestId } = render(<CustomModal isOpen={true} content="test" title="test title" />);
    expect(getByTestId(stringConstants.CustomModal.titleTestId)).toBeTruthy();
  });

  it("doesn't render a title when not asked", async () => {
    const { queryByTestId } = render(<CustomModal isOpen={true} content="test" />);
    expect(queryByTestId(stringConstants.CustomModal.titleTestId)).toBeFalsy();
  });

  it("renders close button when asked", async () => {
    const { getByTestId } = render(<CustomModal isOpen={true} content="test" showCloseButton />);
    expect(getByTestId(stringConstants.CustomModal.closeButtonTestId)).toBeTruthy();
  });

  it("doesn't render close button when not asked", async () => {
    const { queryByTestId } = render(<CustomModal isOpen={true} content="test" />);
    expect(queryByTestId(stringConstants.CustomModal.closeButtonTestId)).toBeFalsy();
  });

  it("renders custom content", async () => {
    const { getByText } = render(<CustomModal isOpen={true} content={<p>Custom Content</p>} />);
    expect(getByText("Custom Content")).toBeTruthy();
  });

  it("renders action buttons", async () => {
    const { getByTestId } = render(
      <CustomModal
        isOpen={true}
        content="test"
        primaryButtonProps={{ testID: "PrimaryButtonTestId", label: "primary button" }}
        secondaryButtonProps={{ testID: "SecondaryButtonTestId", label: "secondary button" }}
      />,
    );
    expect(getByTestId("PrimaryButtonTestId")).toBeTruthy();
    expect(getByTestId("SecondaryButtonTestId")).toBeTruthy();
  });

  it("renders side image", async () => {
    const { getByTestId } = render(
      <CustomModal
        isOpen={true}
        content="test"
        imageProps={{ testID: "ImageTestId", source: imageConstants.Login.loginLeft_image }}
      />,
    );
    expect(getByTestId("ImageTestId")).toBeTruthy();
  });

  it("matches the snapshot", async () => {
    const view = render(
      <CustomModal
        isOpen={true}
        title="Test Title"
        content="Lorem ipsum."
        showCloseButton
        primaryButtonProps={{ label: "Primary Button" }}
        secondaryButtonProps={{ label: "Secondary Button" }}
        imageProps={{ source: imageConstants.Login.loginLeft_image }}
      />,
    );
    expect(view.baseElement).toMatchSnapshot();
  });

  it("renders custom modal with invalid primaryButtonProps", async () => {
    const { getByTestId } = render(
      <CustomModal
        isOpen={true}
        content="test"
        primaryButtonProps={undefined}
        imageProps={{ testID: "ImageTestId", source: imageConstants.Login.loginLeft_image }}
      />,
    );
    expect(getByTestId("ImageTestId")).toBeTruthy();
  });

  it("renders custom modal with array of primaryButtonProps", async () => {
    const { getByTestId } = render(
      <CustomModal
        isOpen={true}
        content="test"
        icon={<></>}
        title={"test-title"}
        primaryButtonProps={[{ label: "Primary Button" }]}
        secondaryButtonProps={[{ label: "Secondary Button" }]}
        imageProps={{ testID: "ImageTestId", source: imageConstants.Login.loginLeft_image }}
      />,
    );
    expect(getByTestId("ImageTestId")).toBeTruthy();
  });

  it("renders custom modal open false", async () => {
    const { container } = render(
      <CustomModal
        isOpen={false}
        content="test"
        icon={<></>}
        title={"test-title"}
        primaryButtonProps={[{ label: "Primary Button" }]}
        secondaryButtonProps={[{ label: "Secondary Button" }]}
        imageProps={{ testID: "ImageTestId", source: imageConstants.Login.loginLeft_image }}
      />,
    );
    expect(container).toBeTruthy();
  });
});
