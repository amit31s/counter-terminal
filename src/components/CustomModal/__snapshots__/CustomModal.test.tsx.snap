// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Modal matches the snapshot 1`] = `
.emotion-17 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  color: #2E1A47;
  width: 24px;
  height: 24px;
}

<body
  tabindex="-1"
>
  <div>
    <div
      class="emotion-0 r-flex-13awgt0"
    />
  </div>
  <div>
    <div
      class="r-animationDuration-1ielgck r-animationKeyframes-xx3c9p r-animationTimingFunction-1uypc71 r-bottom-1p0dtai r-left-1d2f490 r-opacity-6dt33c r-position-1xcajam r-right-zchlnj r-top-ipm5af r-zIndex-sfbmgh"
    >
      <div
        class="r-outlineStyle-1ny4l3l"
        role="presentation"
        tabindex="0"
      />
      <div
        class="emotion-0"
      >
        <div
          aria-modal="true"
          class="emotion-0 r-backgroundColor-1niwhzg r-bottom-1p0dtai r-left-1d2f490 r-position-1xcajam r-right-zchlnj r-top-ipm5af"
        >
          <div
            class="emotion-0 r-flex-13awgt0 r-top-ipm5af"
          >
            <div
              class="emotion-0 r-bottom-1p0dtai r-left-1d2f490 r-opacity-orgf3d r-pointerEvents-12vffkv r-position-u8s1d r-right-zchlnj r-top-ipm5af"
              style="transform: translateY(0px) translateX(0px) scale(1) scaleX(1) scaleY(1) rotate(0deg);"
            >
              <div
                class="emotion-0 r-backgroundColor-kemksi r-bottom-1p0dtai r-cursor-7q8q6z r-left-1d2f490 r-opacity-todpsx r-position-u8s1d r-right-zchlnj r-top-ipm5af r-touchAction-1otgn73"
                tabindex="0"
              />
            </div>
            <div
              class="emotion-0 r-bottom-1p0dtai r-left-1d2f490 r-opacity-orgf3d r-pointerEvents-12vffkv r-position-u8s1d r-right-zchlnj r-top-ipm5af"
              style="transform: translateY(0px) translateX(0px) scale(1) scaleX(1) scaleY(1) rotate(0deg);"
            >
              <span
                data-focus-scope-start="true"
                hidden=""
              />
              <div
                class="emotion-0 r-alignItems-1awozwy r-height-1pi2tsx r-justifyContent-1777fci r-pointerEvents-12vffkv r-width-13qz1uu"
              >
                <div
                  class="emotion-0 r-backgroundColor-14lw9ot r-flexDirection-18u37iz r-width-1y0psj5"
                >
                  <div
                    class="emotion-0 r-alignItems-1oszu61 r-paddingBottom-xd6kpl r-paddingLeft-1m04atk r-paddingRight-1pyaxff r-paddingTop-tskmnb r-width-12kfsgm"
                  >
                    <div
                      class="emotion-0 r-flex-13awgt0 r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
                      style="max-width: 100%;"
                    >
                      <div
                        class="emotion-0 r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                      />
                    </div>
                  </div>
                  <div
                    class="emotion-0 r-alignItems-1oszu61 r-paddingBottom-11rk87y r-paddingLeft-1hxq0ab r-paddingRight-gtt5ci r-paddingTop-fd4yh7 r-width-1euk07n"
                  >
                    <div
                      class="emotion-0"
                    >
                      <div
                        class="emotion-0 r-borderBottomWidth-ndvcnb r-flexDirection-18u37iz r-marginBottom-1ifxtd0 r-paddingBottom-1mdbw0j r-paddingLeft-gy4na3 r-paddingRight-9aemit r-paddingTop-wk8lta"
                      >
                        <div
                          class="emotion-15 r-color-1njgeiw r-flex-13awgt0 r-fontSize-1ra0lkn r-fontStyle-13uqrnb r-fontWeight-b88u0q r-letterSpacing-oxtfae r-lineHeight-nwxazl r-textAlign-fdjqy7"
                          data-testid="TitleTestId"
                          dir="auto"
                        >
                          Test Title
                        </div>
                        <div
                          class="emotion-0 r-alignItems-1awozwy r-cursor-1loqt21 r-height-eu3ka r-justifyContent-1777fci r-touchAction-1otgn73 r-width-1aockid"
                          data-focusvisible-polyfill="true"
                          data-testid="CloseButtonTestId"
                          style="box-shadow: #22d3ee 0px 0px 0px 2px; outline-width: 0px;"
                          tabindex="0"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-17"
                            data-testid="CloseIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="emotion-0 r-paddingBottom-1mdbw0j r-paddingLeft-gy4na3 r-paddingRight-9aemit r-paddingTop-wk8lta"
                      >
                        <div
                          class="emotion-15 r-color-1njgeiw r-fontSize-1x35g6 r-fontStyle-13uqrnb r-fontWeight-16dba41 r-letterSpacing-oxtfae r-lineHeight-2agvir r-textAlign-fdjqy7"
                          dir="auto"
                        >
                          Lorem ipsum.
                        </div>
                      </div>
                    </div>
                    <div
                      class="emotion-0 r-borderTopWidth-13yce4e r-flexDirection-1euycsn r-justifyContent-1h0z5md r-marginTop-1v1z2uz r-paddingBottom-1mdbw0j r-paddingLeft-gy4na3 r-paddingRight-9aemit r-paddingTop-wk8lta"
                      data-testid="ActionButtonContainerTestId"
                    >
                      <div
                        aria-label="Primary Button"
                        class="emotion-21 emotion-0 r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-1q56jx2 r-borderColor-1onh70h r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-yq7r3z r-justifyContent-1777fci r-minHeight-j5ldn5 r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          class="emotion-15 r-color-jwli3a r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
                          dir="auto"
                        >
                          Primary Button
                        </div>
                      </div>
                      <div
                        aria-label="Secondary Button"
                        class="emotion-21 emotion-0 r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-14lw9ot r-borderColor-1onh70h r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-1d6rzhh r-cursor-1loqt21 r-height-yq7r3z r-justifyContent-1777fci r-marginRight-3691iy r-minHeight-j5ldn5 r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          class="emotion-15 r-color-cq6wur r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
                          dir="auto"
                        >
                          Secondary Button
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                data-focus-scope-end="true"
                hidden=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="r-outlineStyle-1ny4l3l"
        role="presentation"
        tabindex="0"
      />
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;
